### PR TITLE
Feature/labx 306/worker profile worker profile integration with backend

### PR DESCRIPTION
**Pull request by alanhanaev**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -133,7 +133,6 @@ src/**/*.js
 !src/content/lib/OpportunitiesContent/OpportunitiesContent.js
 !src/content/lib/ApplicationsAndOffersContent/
 !src/content/lib/ClientProfileContent/
-!src/content/lib/WorkerProfileContent/
 !src/content/lib/CreateJobBoardContent/
 
 !src/partials/
```
Comment (sargonpiraev) [inline .eslintignore]: does all changed files listed in eslintignore?

File: src/api/backend/model/ProfileWorkerModel.js [hunk 2]
```diff
@@ -1,13 +1,90 @@
...
+  }),
+  verifiable: PropTypes.shape({
+    intro: PropTypes.string,
+    pageBackground: PropTypes.string,
+    attachments: PropTypes.arrayOf(PropTypes.instanceOf(AttachmentModel)),
+  }),
+  custom: PropTypes.any,
+  socials: PropTypes.arrayOf(PropTypes.instanceOf(ProfileWorkerSocialModel)),
+  services: PropTypes.arrayOf(PropTypes.instanceOf(ProfileWorkerServiceModel)),
+  employments: PropTypes.arrayOf(PropTypes.instanceOf(ProfileWorkerEmploymentModel)),
```
Comment (sargonpiraev) [inline src/api/backend/model/ProfileWorkerModel.js]: this is  not correct schema
see https://github.com/ChronoBank/LaborX-backend/blob/develop/lib/models/Profile/lib/ProfileWorkerModel.js
client model must be same

File: src/api/backend/model/ProfileWorkerModel.js [hunk 3]
```diff
@@ -1,13 +1,90 @@
...
+        description: item.description,
+        fee: item.fee,
+        minFee: item.minFee,
+        attachments: item.attachments ? item.attachments.map(item => new AttachmentModel(item)) : [],
+      })),
+      employments: data[nameField].employments.map(item => new ProfileWorkerEmploymentModel(item)),
+    })}
+    else
+    {return new ProfileWorkerModel({})}
+  }
```
Comment (sargonpiraev) [inline src/api/backend/model/ProfileWorkerModel.js]: this is  not correct schema
see https://github.com/ChronoBank/LaborX-backend/blob/develop/lib/models/Profile/lib/ProfileWorkerModel.js
client model must be same

File: src/api/backend/model/ProfileWorkerModel.js [hunk 4]
```diff
@@ -1,13 +1,90 @@
...
+      intro: "",
+      pageBackground: "",
+      attachments: [],
+    },
+    custom: null,
+    socials: [],
+    services: [],
+    employments: [],
+  }, props)
+}
```
Comment (sargonpiraev) [inline src/api/backend/model/ProfileWorkerModel.js]: this is  not correct schema
see https://github.com/ChronoBank/LaborX-backend/blob/develop/lib/models/Profile/lib/ProfileWorkerModel.js
client model must be same

File: src/api/backend/model/VerificationRequestWorkerModel.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    //     attachments: null
+    //   })
+    //   ),
+    //   employments: form.experiences.map(item => new VerificationRequestWorkerEmploymentModel({
+    //     organization: item.organisation,
+    //     since: item.workFrom,
+    //     until: item.workTo,
+    //     responsibilities: item.responsibilities
+    //   }))
+    // });
```
Comment (sargonpiraev) [inline src/api/backend/model/VerificationRequestWorkerModel.js]: del

File: src/components/common/Dialog/Dialog.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Dialog from 'material-ui/Dialog'
+import { MuiThemeProvider } from 'material-ui/styles'
```
Comment (sargonpiraev) [inline src/components/common/Dialog/Dialog.js]: use Dialog from @materail-ui/core
MuiThemeProvider uses in _app.js
no need to duplicate it here

File: src/content/lib/DashboardContent/DashboardContent.js [hunk 7]
```diff
@@ -38,7 +38,7 @@ export class DashboardContent extends React.Component {
                   secondIcon: Image.SETS.MESSAGE_ERROR,
                 },
                 ...(this.props.user.accountTypes.client ? [ { href: '/client-profile', label: 'nav.clientProfile', isLink: true, } ] : []),
-                ...(this.props.user.accountTypes.worker ? [ { href: '/worker-profile', label: 'nav.workerProfile', isLink: true, } ] : []),
+                ...( /*this.props.user.accountTypes.worker */ true ? [ { href: '/worker-profile', label: 'nav.workerProfile', isLink: true, } ] : []),
```
Comment (sargonpiraev) [inline src/content/lib/DashboardContent/DashboardContent.js]: ?

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 8]
```diff
@@ -76,33 +91,54 @@ export default class ServicesTab extends React.Component {
...
+              <div key={item.id} className={css.attachment}>
+                <div className={css.wrapAttachmentName}>
+                  {item.name}
+                </div>
+                <div className={css.wrapDeleteAttachmentBtn} onClick={() => onDeleteServiceAttachment(item.id)}>
+                  <Icon
+                    icon={Icon.ICONS.DELETE}
+                    color={Icon.COLORS.GREY30}
+                    size={20}
+                  />
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: use https://material-ui.com/api/svg-icon/ instead 
https://material.io/tools/icons/?icon=delete&style=baseline

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 9]
```diff
@@ -76,33 +91,54 @@ export default class ServicesTab extends React.Component {
...
-            size={28}
-          />
-          <p>Upload service agreement</p>
+          <label className={css.fileLoaderBlock}>
+            <input type='file' style={{ display: 'none' }} onChange={(e) => onUploadServiceAgreement(e, index)} multiple={false} />
+            <Icon
+              icon={Icon.ICONS.UPLOAD}
+              color={Icon.COLORS.BLUE}
+              size={28}
+            />
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: use https://material-ui.com/api/svg-icon/ instead
https://material.io/tools/icons/?icon=cloud_upload&style=baseline

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 10]
```diff
@@ -127,80 +163,21 @@ export default class ServicesTab extends React.Component {
           />
         </div>
         <FieldArray name='services' component={this.renderServices} />
-        <div className={css.twoColumn}>
+        <div className={""}>
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: ?

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 11]
```diff
@@ -127,80 +163,21 @@ export default class ServicesTab extends React.Component {
...
-                fullWidth
-                component={TextField}
-                name='hourlyChargeUsd'
-                hintText='$0.00'
-                floatingLabelStyle={floatStyle}
-                floatingLabelText='empty'
-                floatingLabelFixed
-              />
-            </div>
+            {/* <div className={css.twoColumn}> */}
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: ?

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 12]
```diff
@@ -215,11 +192,11 @@ export default class ServicesTab extends React.Component {
             name='currencyBitcoin'
             label='Bitcoin'
           />
-          <Field
+          {/* <Field
             component={ValidatedCheckbox}
             name='currencyAnother'
             label='Another Currency'
-          />
+          /> */}
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: ?

File: src/content/lib/WorkerProfileContent/WorkExperienceTab/WorkExperienceTab.js [hunk 13]
```diff
@@ -106,78 +120,83 @@ export default class WorkExperienceTab extends React.Component {
...
+                  size={28}
+                />
+                <p>Upload Certificate(s)</p>
+              </div>
+              <div className={css.documentEntry}>
+                <Icon
+                  icon={Icon.ICONS.UPLOAD}
+                  color={Icon.COLORS.BLUE}
+                  size={28}
+                />
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/WorkExperienceTab/WorkExperienceTab.js]: better to use @material-ui/icons with SvgIcon component

File: src/content/lib/WorkerProfileContent/WorkerProfileContent.js [hunk 14]
```diff
@@ -46,12 +55,78 @@ class WorkerProfileContent extends React.Component {
 
   handleClickAddWorker = () => {
     // eslint-disable-next-line no-console
-    console.log('---WorkerProfileContent handleClickAddWorker')
+    if (this.state.slideIndex === 1)
+    {this.props.addEmptyWorkerExperience()}
+    if (this.state.slideIndex === 2)
+    {this.props.addEmptyWorkerService()}
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/WorkerProfileContent.js]: we do not use this notation

use

```
if (this.state.slideIndex === 1) {
  this.props.addEmptyWorkerExperience()
}
if (this.state.slideIndex === 2) {
  this.props.addEmptyWorkerService()
}
```

or 

```
if (this.state.slideIndex === 1) this.props.addEmptyWorkerExperience()
if (this.state.slideIndex === 2) this.props.addEmptyWorkerService()
```

File: src/content/lib/WorkerProfileContent/WorkerProfileContent.js [hunk 15]
```diff
@@ -104,13 +180,14 @@ class WorkerProfileContent extends React.Component {
             ) : null }
           </div>
           <div className={css.tabContent}>
+          {/* {JSON.stringify(this.props.serviceAttachments)} */}
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/WorkerProfileContent.js]: del

File: src/store/worker-profile/actions.js [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+    const state = getState()
+    const token = userTokenSelector()(state)
+    const res = await backendApi.submitWorkerProfile(workerProfile, token)
+    const initialValues = getWorkerProfileInitialValues(res.profile)
+    dispatch(initialize(WORKER_PROFILE_FORM, initialValues))
+    dispatch(reviewWorkerProfileSuccess(res.profile))
+  } catch (err) {
+    dispatch(reviewWorkerProfileFailure(err))
+  }
+}
```
Comment (sargonpiraev) [inline src/store/worker-profile/actions.js]: this action calls backendApi.submitWorkerProfile but name is reviewWorkerProfile
this action should named submitWorkerProfile

File: src/store/worker-profile/actions.js [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+    data.experiences.forEach((element) => {
+      experiences.push(new VerificationRequestWorkerEmploymentModel({
+        organization: element.organization,
+        since: element.since,
+        until: element.until,
+        responsibilities: element.responsibilities
+      }))
+    });
+  return experiences;
+}
```
Comment (sargonpiraev) [inline src/store/worker-profile/actions.js]: all this getters should be placed in selectors.js

**Review by sargonpiraev — CHANGES_REQUESTED** [review]

File: src/content/lib/WorkerProfileContent/GeneralTab/GeneralTab.js [hunk 18]
```diff
@@ -1,36 +1,46 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { Field } from 'redux-form'
-import { TextField, SelectField } from 'redux-form-material-ui'
-import { MenuItem } from 'material-ui'
-import { Icon } from 'src/components/common'
+import { TextField } from 'redux-form-material-ui'
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/GeneralTab/GeneralTab.js]: import { TextField } from 'redux-form-material-ui-next'

we use new components

File: src/components/common/Dialog/Dialog.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Dialog from 'material-ui/Dialog'
```
Comment (sargonpiraev) [inline src/components/common/Dialog/Dialog.js]: use new material ui

File: src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js [hunk 20]
```diff
@@ -127,99 +129,34 @@ export default class ServicesTab extends React.Component {
           />
         </div>
         <FieldArray name='services' component={this.renderServices} />
-        <div className={css.twoColumn}>
+        <div className=''>
```
Comment (sargonpiraev) [inline src/content/lib/WorkerProfileContent/ServicesTab/ServicesTab.js]: ?

File: src/store/worker-profile/actions.js [hunk 21]
```diff
@@ -0,0 +1,127 @@
...
+    const state = getState()
+    const token = userTokenSelector()(state)
+    const res = await backendApi.submitWorkerProfile(workerProfile, token)
+    const initialValues = getWorkerProfileInitialValues(res.profile)
+    dispatch(initialize(WORKER_PROFILE_FORM, initialValues))
+    dispatch(submitWorkerProfileSuccess(res.profile))
+  } catch (err) {
+    dispatch(submitWorkerProfileFailure(err))
+  }
+}
```
Comment (sargonpiraev) [inline src/store/worker-profile/actions.js:28]: you renamed action creator but not action names
use WORKER_PROFILE_SUBMIT

File: src/content/lib/GeneralProfileContent/ContactsFormModel.js [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+    const profile = submitted || approved
+    return new ProfileContactsFormModel(
+      profile ? {
+        email: profile.email,
+        phone: profile.phone,
+      } : {}
+    )
+  }
+
+}
```
Comment (sargonpiraev) [inline src/content/lib/GeneralProfileContent/ContactsFormModel.js]: what this model is for?

File: src/components/common/Dialog/Dialog.js [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+
+    return (
+        <Dialog onClose={onClose} {...other}>
+          <div>
+            {children}
+          </div>
+        </Dialog>
+    )
+  }
+}
```
Comment (sargonpiraev) [inline src/components/common/Dialog/Dialog.js]: where this component used?

File: src/api/backend/model/ProfileWorkerModel.js [hunk 24]
```diff
@@ -1,13 +1,87 @@
...
+    else {
+      return VALIDATION_STATE.INITIAL
+    }
+  }
+
+  static getValidationComment (profile) {
+    if (profile)
+    {return profile.submitted && profile.submitted.validationComment}
+    else
+    {return ""}
```
Comment (sargonpiraev) [inline src/api/backend/model/ProfileWorkerModel.js]: we do not use this notation

**Review by sargonpiraev — CHANGES_REQUESTED** [review]

File: src/components/common/Dialog/Dialog.js (hunk 23 above)
Comment (alanhanaev) [reply src/components/common/Dialog/Dialog.js]: strange, did not create it. Perhaps copied from the old version of develop

File: src/api/backend/model/ProfileWorkerModel.js (hunk 24 above)
Comment (alanhanaev) [reply src/api/backend/model/ProfileWorkerModel.js]: Okay, sorry. Inserted by mistake, the code formatting function in the vs code editor

File: src/content/lib/GeneralProfileContent/ContactsFormModel.js (hunk 22 above)
Comment (alanhanaev) [reply src/content/lib/GeneralProfileContent/ContactsFormModel.js]: strange, Assume that it took from the old version of the development branch, after an unsuccessful merger

**Review by sargonpiraev — APPROVED** [review]

**Pull request merged**